src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -240,47 +250,64 @@ public void run() {
-
+    void maybeAddPartitions(int minPartitionNum) throws ExecutionException, InterruptedException {
       try {
-        scala.collection.Map<Object, scala.collection.Seq<Object>> existingAssignment = getPartitionAssignment(zkClient, _topic);
-        int partitionNum = existingAssignment.size();
-
+        Collection<String> topicNames = _adminClient.listTopics().names().get();
+        Map<String, KafkaFuture<TopicDescription>> kafkaFutureMap = _adminClient.describeTopics(topicNames).values();
+        KafkaFuture<TopicDescription> topicDescriptions = kafkaFutureMap.get(_topic);
+        List<TopicPartitionInfo> partitions = topicDescriptions.get().partitions();
+        int partitionNum = partitions.size();
         if (partitionNum < minPartitionNum) {
-          LOG.info("MultiClusterTopicManagementService will increase partition of the topic {} "
-              + "in cluster {} from {} to {}.", _topic, _zkConnect, partitionNum, minPartitionNum);
+          LOG.info("{} will increase partition of the topic {} in the cluster {} from {}"
+              + " to {}.", this.getClass().toString(), _topic, _zkConnect, partitionNum, minPartitionNum);
           Set<Integer> blackListedBrokers =
               _topicFactory.getBlackListedBrokers(_zkConnect);
-          scala.Option<scala.collection.Map<java.lang.Object, scala.collection.Seq<java.lang.Object>>> replicaAssignment = scala.Option.apply(null);
-          scala.Option<Seq<Object>> brokerList = scala.Option.apply(null);
-          Set<BrokerMetadata> brokers =
-              new HashSet<>(scala.collection.JavaConversions.asJavaCollection(adminZkClient.getBrokerMetadatas(RackAwareMode.Disabled$.MODULE$, brokerList)));
-
+          Set<BrokerMetadata> brokers = new HashSet<>();
+          Iterator<Node> nodesIterator = _adminClient.describeCluster().nodes().get().iterator();
+          while (nodesIterator.hasNext()) {
+            KafkaFuture<Collection<Node>> clusterNodes = _adminClient.describeCluster().nodes();
+            BrokerMetadata brokerMetadata = new BrokerMetadata(
+                clusterNodes.get().iterator().next().id(), null
+            );
+            brokers.add(brokerMetadata);
+          }
           if (!blackListedBrokers.isEmpty()) {
             brokers.removeIf(broker -> blackListedBrokers.contains(broker.id()));
           }
-          adminZkClient.addPartitions(_topic, existingAssignment,
-              scala.collection.JavaConversions.collectionAsScalaIterable(brokers).toSeq(),
-              minPartitionNum, replicaAssignment, false);
+          Map<String, NewPartitions> newPartitionsMap = new HashMap<>();
+          NewPartitions newPartitions = NewPartitions.increaseTo(minPartitionNum, null);
+          newPartitionsMap.put(_topic, newPartitions);
+          _adminClient.createPartitions(newPartitionsMap);
         }
       } finally {
-        zkClient.close();
+        _adminClient.close();
```

Comment:
This seems fine as long as a new Helper class is instantiated before every method is called.  Is that the case?  Otherwise I'm not sure what happens if a closed adminClient is reused.  Probably nothing good.